### PR TITLE
add solution to the medium replace first challenge

### PR DESCRIPTION
**Pull request by kerolossamir165**
# Challenge [ Medium Replace First ]

## Checklist

- [x] Make sure that you are opening the PR with a single solution (one solution per PR)
- [x] The solution must have at least several references for the readers to dive in depth

### CODE REVIEW

**Comment by kerolossamir165** [comment]
@ghaiklor could you please review this PR 
Thanks

**Comment by ghaiklor** [comment]
@kerolossamir165 please, do a grammar check using LanguageTool, for example.

File: en/medium-replcaefirst.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+id: 25170
+title: ReplaceFirst
+lang: en
+level: medium
+tags: infer array
+---
+
+## Challenge
+
+Implement the type ReplaceFirst<T, S, R> which will replace the first occurrence of S in a tuple T with R. If no such S exists in T, the result should be T.
```
Comment (ghaiklor) [inline en/medium-replcaefirst.md:11]: Use quoted text to highlight code

File: en/medium-replcaefirst.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+---
```
Comment (ghaiklor) [inline en/medium-replcaefirst.md:1]: You have an error in filename

Comment (kerolossamir165) [reply en/medium-replcaefirst.md:1]: sorry for that, i will change them all and push again

**Review by ghaiklor — APPROVED** [review]

**Pull request merged**